File: Day-1/Exchanging-Gifts/main.py
def find_youngest_member(n, m, gifts):
    if m == 0:
         return n
    table = {}
    for gift in gifts:
        if gift[1] in table:
            table[gift[1]] += 1
        else:
            table[gift[1]] = 1

    table = dict(sorted(table.items(), key=lambda item: item[1], reverse=True))
    lst = list(table.items())
    
    if lst[0][1] == n - 1:
        return lst[0][0]
    return -1

File: Day-1/Exchanging-Gifts/Main.java
import java.util.HashMap;
import java.util.Map;

public class Main {

    public static void find_youngest_member(int n, int m, int[][] gifts) {
        if (m == 0) {
            System.out.println(n);
            return;
        }
        Map<Integer, Integer> freq = new HashMap<>();

        for (int[] gift : gifts) {
            freq.merge(gift[1], 1, Integer::sum);
        }

        for (var entry : freq.entrySet()) {
            if (entry.getValue() == n - 1) {
                System.out.println(entry.getKey());
                return;
            }
        }
        System.out.println(-1);
    }
}